### PR TITLE
Fix: leading comments added from previous node (fixes #256)

### LINKED ISSUE #256: Incorrect leading comments.
From https://github.com/eslint/eslint/pull/5512

In the following case:

```js
var foo = {
  // foo
};
var bar;
```

`// foo` comment has attached to the leading of `var bar`.

----

```js
"use strict";

const espree = require("espree");
const ast = espree.parse(
    `var foo = {
  // foo
};
var bar;`,
    {
        range: true,
        attachComment: true
    }
);

console.log(JSON.stringify(ast, null, 4));
```

```json
{
    "type": "Program",
    "start": 0,
    "end": 32,
    "range": [
        0,
        32
    ],
    "body": [
        {
            "type": "VariableDeclaration",
            "start": 0,
            "end": 23,
            "range": [
                0,
                23
            ],
            "declarations": [
                {
                    "type": "VariableDeclarator",
                    "start": 4,
                    "end": 22,
                    "range": [
                        4,
                        22
                    ],
                    "id": {
                        "type": "Identifier",
                        "start": 4,
                        "end": 7,
                        "range": [
                            4,
                            7
                        ],
                        "name": "foo"
                    },
                    "init": {
                        "type": "ObjectExpression",
                        "start": 10,
                        "end": 22,
                        "range": [
                            10,
                            22
                        ],
                        "properties": [],
                        "trailingComments": [
                            {
                                "type": "Line",
                                "value": " foo",
                                "start": 14,
                                "end": 20,
                                "range": [
                                    14,
                                    20
                                ]
                            }
                        ]
                    }
                }
            ],
            "kind": "var"
        },
        {
            "type": "VariableDeclaration",
            "start": 24,
            "end": 32,
            "range": [
                24,
                32
            ],
            "declarations": [
                {
                    "type": "VariableDeclarator",
                    "start": 28,
                    "end": 31,
                    "range": [
                        28,
                        31
                    ],
                    "id": {
                        "type": "Identifier",
                        "start": 28,
                        "end": 31,
                        "range": [
                            28,
                            31
                        ],
                        "name": "bar"
                    },
                    "init": null
                }
            ],
            "kind": "var",
            "leadingComments": [
                {
                    "type": "Line",
                    "value": " foo",
                    "start": 14,
                    "end": 20,
                    "range": [
                        14,
                        20
                    ]
                }
            ]
        }
    ],
    "sourceType": "script",
    "comments": [
        {
            "type": "Line",
            "value": " foo",
            "start": 14,
            "end": 20,
            "range": [
                14,
                20
            ]
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by kaicataldo**
Looks like the `mootools` test is failing. Not sure if this means my change is bad or the expected output from the test is incorrect now that this change has been made. Are these generated with a command somewhere?

I ran this version of my change against the current master branch to test for speed and was surprised to find no conclusive evidence of this slowing things down (most likely because my change only runs on comments and only when the node type is `Program`)...Some tests were a bit slower, and others were a bit faster. So doesn't seem like a huge impact on performance :+1:

### CODE REVIEW

**Comment by nzakas** [comment]
Can you add a test for this change?

Chances are that there was an error in Mootools output that this fixes. You can run `node tools/update-tests.js` to update, though you should revert any other changes that tool makes as it causes some errors in other tests (haven't gotten around to fixing that).

**Comment by kaicataldo** [comment]
Yeah, of course! Sorry, wasn't understanding how these tests were organized/generated, but I see now. Let me know if I can do anything else!

May take a look at #212 after this PR lands, since I've familiarized myself with the code and the change is probably a similar thing :)

File: tests/fixtures/attach-comments/comment-within-previous-node.result.js [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+                                    },
+                                    "range": [
+                                        21,
+                                        26
+                                    ],
+                                    "value": "bar",
+                                    "raw": "'bar'"
+                                },
+                                "kind": "init",
+                                "trailingComments": [
```
Comment (kaicataldo) [inline tests/fixtures/attach-comments/comment-within-previous-node.result.js]: Correctly adds trailing comment to `bar` property

File: tests/fixtures/attach-comments/comment-within-previous-node.result.js [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+                    "column": 8
+                }
+            },
+            "range": [
+                46,
+                54
+            ],
+            "declarations": [
+                {
+                    "type": "VariableDeclarator",
```
Comment (kaicataldo) [inline tests/fixtures/attach-comments/comment-within-previous-node.result.js]: No leading comments!

**Comment by nzakas** [comment]
Awesome, thank you!

**Pull request merged**